packages/repository-json-schema/src/build-schema.ts

```diff
@@ -42,6 +42,11 @@ export interface JsonSchemaOptions<T extends object> {
    */
   optional?: (keyof T)[];
 
+  /**
+   * Boolean to generate fieldNames maximizing compatibility (Azure)
+   */
+  itExtendsCompatibility?: boolean;
```

Comment:
I think we should just fix the title with compatible chars without introducing an extra flag. We should also populate `description` to have more meaningful details.

Reply:
I added that flag to give the opportunity to the users to keep the easy to understand format of 'Exclude|Optional[key1,key2...]' that you designed, but if you prefer to keep it simple and leave always a compatible format, ok! I'll check the description too.

Reply:
@raymondfeng please, check the current status of this PR and decide which option to take:
- I keep along with this flag
or
- apply the compatible format without flag, loosing expressivity

Reply:
I personally think we should keep this as a flag (first option) because the way it is right now is more readable.

Reply:
What I propose is as follows:

1. Make schema `title` compatible by default with a bit less expressivity (IIRC, the `title` needs to be unique)
2. Populate schema `description` to make it readable

Reply:
Personally, I am against too many feature flags (see [Decision Fatique](https://en.wikipedia.org/wiki/Decision_fatigue)). IMO, a good tool should pick a reasonable behavior so that users don't have to learn about different options to be able to make a qualified decision.

Let's follow Raymond's proposal.

>1. Make schema `title` compatible by default with a bit less expressivity (IIRC, the `title` needs to be unique)
> 2. Populate schema `description` to make it readable